AWS/DirectoryInsights/get-jcdirectoryinsights.py

```diff
@@ -0,0 +1,113 @@
+import requests, datetime, json, boto3, os, gzip
+from botocore.exceptions import ClientError
+
+def get_secret(secret_name):
+    client = boto3.client(service_name='secretsmanager')
+    try:
+        get_secret_value_response = client.get_secret_value(SecretId=secret_name)
+    except ClientError as e:
+        raise Exception(e)
+        
+    secret = get_secret_value_response['SecretString']
+    return secret
+
+def jc_directoryinsights(event, context):
+    try:
+        jcapikeyarn = os.environ['JcApiKeyArn']
+        incrementType = os.environ['incrementType']
+        incrementAmount = int(os.environ['incrementAmount'])
+        bucketName = os.environ['BucketName']
+    except KeyError as e:
+        raise Exception(e)
+
+    jcapikey = get_secret(jcapikeyarn)
```

Comment:
This looks right. You might need to add region to that. If you want to pull the same region as the lambda you can access that from an environment var. Your current region maybe defaulted. 

    region = os.environ.get("AWS_REGION")

I would usually use a key value or a json object. Then you can pull out and use what you what from that. For just one key I think this works.